module_utils/storage_lsr/argument_validator.py

```diff
@@ -0,0 +1,171 @@
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+
+class ArgValidator(object):
+
+    @classmethod
+    def _validate_list(cls, value):
+        if isinstance(value, list):
+            return value
+
+        if isinstance(value, str):
+            return value.split(",")
+
+        if isinstance(value, int) or isinstance(value, float):
+            return [value]
+
+        raise TypeError("'%s' has to be a list" % value)
+
+    @classmethod
+    def _validate_str(cls, value):
+        return str(value)
+
+    @classmethod
+    def _validate_bool(cls, value):
+        BOOLEANS_TRUE = ['y', 'yes', 'on', '1', 'true', 't', 1, 1.0, True]
+        BOOLEANS_FALSE = ['n', 'no', 'off', '0', 'false', 'f', 0, 0.0, False]
```

Comment:
Do you really want or need to support all of these different values?  Note that Ansible/Yaml will convert values `value: off` to python `False` - and if someone explicitly specifies something like `value: "off"` meaning the string value "off" instead of the boolean value - I think that should (usually) be an error.
Also note that there is the function https://docs.python.org/3/distutils/apiref.html#distutils.util.strtobool which can handle most of these - and I don't think you need to worry about `float` `0.0` and `1.0`